build_projects/dotnet-cli-build/GenerateBuildVersionInfo.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.DotNet.Cli.Build
 {
-    public class GenerateBuildVersionInfo : Task
+    public class GenerateBuildVersionInfo : ToolTask
```

Comment:
It seems strange that this is a tool task. This task wraps a combination of logic that ends in a generated version info, and not only a particular tool invocation, git in this case. 

What is the reasoning for making this a tool task?